### PR TITLE
Migrate to CircleCI 2

### PR DESCRIPTION
**Pull request by autrilla**
Builds on my fork (https://circleci.com/gh/autrilla/tokenserver/15, disregard docker push errors).

### CODE REVIEW

File: bin/ci/deploy-dockerhub.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# THIS IS MEANT TO BE RUN BY CI
+
+set -e
```
Comment (rfk) [inline bin/ci/deploy-dockerhub.sh]: from https://github.com/mozilla-services/syncserver/pull/117#discussion_r186092204, should we add `-x` here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            app:build /bin/sh -c "nosetests tokenserver/tests"
+      - run:
+          name: Push to Dockerhub
+          command: |
+            # set DOCKER_DEPLOY=true in Circle UI to push to Dockerhub
+            DOCKER_DEPLOY="${DOCKER_DEPLOY:-false}"
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+              bin/ci/deploy-dockerhub.sh latest
+            fi
+            if [[ "${CIRCLE_BRANCH}" == dockerpush* ]]; then
```
Comment (rfk) [inline .circleci/config.yml]: For consistency with other repos, could we please add `[[ "${CIRCLE_BRANCH}" == feature* ]]` here as well?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          command: >-
+            docker run
+            --link mysql:mysql
+            --env "MOZSVC_SQLURI=pymysql://ubuntu:ubuntu@mysql/tokenserver_test"
+            app:build /bin/sh -c "nosetests tokenserver/tests"
+      - run:
+          name: Push to Dockerhub
+          command: |
+            # set DOCKER_DEPLOY=true in Circle UI to push to Dockerhub
+            DOCKER_DEPLOY="${DOCKER_DEPLOY:-false}"
```
Comment (rfk) [inline .circleci/config.yml]: IIUC from https://github.com/mozilla-services/syncserver/pull/117#discussion_r186093979 we decided this `DOCKER_DEPLOY` variable wasn't necessary?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Install essential packages
+          command: |
+            apt-get update
+            apt-get install -y ca-certificates curl build-essential make git
+            curl -L -o /tmp/dockerize.tgz https://github.com/jwilder/dockerize/releases/download/v0.6.1/dockerize-linux-amd64-v0.6.1.tar.gz
+            tar -xz -C /usr/bin -f /tmp/dockerize.tgz
```
Comment (rfk) [inline .circleci/config.yml]: Do I understand correctly, that this is only necessary for running the (separate) database image used for testing the app, and doesn't actually go into the production build?

Comment (autrilla) [reply .circleci/config.yml]: Correct. I've added a comment to clarify this.

File: .circleci/config.yml (hunk 3 above)
Comment (autrilla) [reply .circleci/config.yml]: Yep, I've removed it

File: .circleci/config.yml (hunk 2 above)
Comment (autrilla) [reply .circleci/config.yml]: Done

File: bin/ci/deploy-dockerhub.sh (hunk 1 above)
Comment (autrilla) [reply bin/ci/deploy-dockerhub.sh]: Done

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+          name: Test nose
+          command: >-
+            docker run
+            --link mysql:mysql
+            --env "MOZSVC_SQLURI=pymysql://ubuntu:ubuntu@mysql/tokenserver_test"
+            app:build /bin/sh -c "nosetests tokenserver/tests"
+      - run:
+          name: Push to Dockerhub
+          command: |
+            # set DOCKER_DEPLOY=true in Circle UI to push to Dockerhub
```
Comment (rfk) [inline .circleci/config.yml]: `DOCKER_DEPLOY` still appears in a comment here, and at the top of the file.

**Comment by rfk** [comment]
I fixed a couple of nits over in https://github.com/mozilla-services/tokenserver/pull/119, will merge there assuming it passes CI.  Thanks @autrilla!

**Comment by vladikoff** [comment]
Seems to have a syntax error

**Comment by autrilla** [comment]
Yeah. There's also some things from https://github.com/mozilla-services/server-syncstorage/pull/81 that I should bring in over here.

**Comment by rfk** [comment]
Closing this and updating https://github.com/mozilla-services/tokenserver/pull/119

**Pull request closed without merging**